[blog post] astro-i18n (#802)
withastro/astro.build: src/content/blog/astro-i18n.mdx

Reviewer: More of an "open suggestion": perhaps there's a way to fit the release with so many translations from day one into this? - Might be something to mention only on a Twitter post or whatever, but would tie in nicely with the launch week theme (community i18n, starlight, 3.0)...

Author: I think this makes a good social media post! Chris made one on Mastodon already. We'll plan one like that for tomorrow!

---
title: "i18n at Astro"
description: "i18n efforts at Astro - update me!"
publishDate: "August 31, 2023"
authors:
  - yan
  - sarah

coverImage: "/src/content/blog/_images/astro-i18n/blog-hero-post-header.webp"
socialImage: "/src/content/blog/_images/astro-i18n/blog-social.webp"
lang: "en"
---

import BlogContentImage from "/src/components/BlogContentImage.astro"
import translatorBadges from "/src/content/blog/_images/astro-i18n/translator-badges.webp"

**Welcome, World.**

**“Astro is for everyone”** is a common saying on our team. It began as a saying to help us design APIs and relate to our users. But as Astro has evolved, this core value of accessibility has worked its way into more than just feature development.

## Astro is for everyone

If Astro is truly meant to be for everyone, then you must include the majority of the planet (83%) that does not speak English! 18 months ago, we decided to get serious and invest in our overhauling our documentation, and this included a major internationalization (i18n) initiative.

> Today, Astro docs has been officially translated into 12 different languages with over 1000 pages translated in total. Even our full introductory tutorial (32 pages in total) has been translated into six different languages. These languages cover billions of people across more than 50 countries, opening the door to many who would otherwise struggle to use Astro reliably.

Author: Starlight has only been translated into 9 docs, and we didn't choose Starlight the same way the others, so I *think* we'll leave this, at least here in the intro! I might work Starlight in a bit later!

And of course, you'll also find them dropping docs links and answering questions in our `#support` threads because they know the docs *so well* from having translated them!

## Automation and Tooling

It’s one thing to know *how* to get started. But, translators also need to know *where* to get started!

Led by beloved Astro maintainer Hippo, our docs team built a custom [Translation Tracker](https://i18n.docs.astro.build/), the showpiece of our i18n automation! It tracks every docs page marked as "available to translate" by language and reports its own current state: 

- fully-translated and up-to-date;
- needing an update to match the English documentation;
- or, not yet translated.

<BlogContentImage src={trackerTable} alt="" />

Internally, it uses the git history to compare changes between languages, defines the status of each page through some checks, and also generates a GitHub link with all the commits a page needs to be in sync with the English version. It’s powered by GitHub Actions, right out of our docs repo.

<BlogContentImage src={trackerSummary} alt="" />

> Our Translation Tracker also uses GitHub data to list open translation PRs in need of reviews from native speakers, which are a great way to get started without the pressure of opening your own PR!

Reviewer: I guess we're referring only to the Tracker here? So it should be "is" instead? Feel free to ignore if I am wrong!
```suggestion
Our Translation Tracker also uses GitHub data to list open translation PRs in need of reviews from native speakers, which is a great way to get started without the pressure of opening your own PR!
```

Author: We're referring to open PRs, They ARE a great way to get started!

<BlogContentImage src={translationReviews} alt="" />

**With a clear picture of what our translation needs are, we are able to quickly welcome and onboard new i18n contributors who already know which contributions are wanted.**

## Driving decisions through data

Initially, we made translation decisions (e.g. which languages to support in docs) based solely on our community’s excitement and availability to help us. This proved to be a good strategy when we were just getting started. But as Astro docs grew, we needed to make more informed decisions.

> One of the difficult calls we had to make early on was to decide which languages we would officially support with translations. Or rather, which ones we would not. Officially supporting a language is quite a large committment, which unfortunately means that we cannot accept every language proposed to us. Our goal was to find a balance between languages easier for us to maintain and the ones that would make the most impact. Access to quality information and education isn’t the same everywhere — and as the ones with the privilege of English proficiency ourselves, we are in a position of strength to address these inequities and advocate for the communities who need translations the most.

Reviewer: Too many t's
```suggestion
One of the difficult calls we had to make early on was to decide which languages we would officially support with translations. Or rather, which ones we would not. Officially supporting a language is quite a large commitment, which unfortunately means that we cannot accept every language proposed to us. Our goal was to find a balance between languages easier for us to maintain and the ones that would make the most impact. Access to quality information and education isn’t the same everywhere — and as the ones with the privilege of English proficiency ourselves, we are in a position of strength to address these inequities and advocate for the communities who need translations the most.
```

To help us make these difficult decisions, we analyzed several data points such as the number of members for each language in our Discord, a list of countries per language, demographics of several developer surveys, and [EF’s English Proficiency Index](https://www.ef.com/wwpt/epi/). By starting with clear goals, we could choose languages that helped keep us focused on meeting the needs of the global developer community.

## Supporting our contributors

> The last couple of months in particular have been dedicated to supporting and recognizing the massive work done by our internationalization teams. While Astro Docs has long had a “Face Pile” of the GitHub avatars of every contributor to the repository, we now display a [dedicated translators’ face pile](https://docs.astro.build/es/getting-started/#traducir) on each translated version of the site. Our popular community [Astro Badges](https://astro.badg.es/v1/achievements/) also now display dedicated contributions to acknowledge both translations and translation reviews.

Reviewer: We have "dedicated" three times here, maybe we can diversify this a bit? I was thinking something like this, but feel free to run with whatever makes more sense for you!
```suggestion
The last couple of months in particular have been focused on supporting and recognizing the massive work done by our internationalization teams. While Astro Docs has long had a “Face Pile” of the GitHub avatars of every contributor to the repository, we now display a [dedicated translators’ face pile](https://docs.astro.build/es/getting-started/#traducir) on each translated version of the site. Our popular community [Astro Badges](https://astro.badg.es/v1/achievements/) also now displays unique i18n contributions to acknowledge both translations and translation reviews.
```

<BlogContentImage src={translatorBadges} alt="Four Astro badges, displaying various achievement levels for community members, such as Universal Translator for merging 40 i18n PRs or Rosetta Stone for reviewing 40 i18n PRs" />

> Thanks to the Astro open-source project having so many generous sponsors and backers, we are able to issue regular monetary Community Awards. At the time of posting, we have already given more than $4000 Open Collective funds directly to our international community. Other forms of rewards include limited edition swag items from the [Astro Swag Shop](https://shop.astro.build/) (all proceed go directly back into our Open Collective), stipends and sponsorships.

Reviewer: Oxford comma and `proceed` -> `proceeds`
```suggestion
Thanks to the Astro open-source project having so many generous sponsors and backers, we are able to issue regular monetary Community Awards. At the time of posting, we have already given more than $4000 Open Collective funds directly to our international community. Other forms of rewards include limited edition swag items from the [Astro Swag Shop](https://shop.astro.build/) (all proceeds go directly back into our Open Collective), stipends, and sponsorships.
```

---

> title: "i18n at Astro"
> description: "i18n efforts at Astro - update me!"

Author: ```suggestion
title: "How Astro does i18n"
description: "Leading open-source documentation with and for an international community."
```

publishDate: "August 31, 2023"
authors:
  - yan
  - sarah

coverImage: "/src/content/blog/_images/astro-i18n/blog-hero-post-header.webp"
socialImage: "/src/content/blog/_images/astro-i18n/blog-social.webp"
lang: "en"
---

import BlogContentImage from "/src/components/BlogContentImage.astro"
import translatorBadges from "/src/content/blog/_images/astro-i18n/translator-badges.webp"
import trackerSummary from "/src/content/blog/_images/astro-i18n/translation-tracker-summary.webp"
import trackerTable from "/src/content/blog/_images/astro-i18n/translation-tracker-table.webp"
import translationReviews from "/src/content/blog/_images/astro-i18n/translations-needing-reviews.webp"
import langPicker from "/src/content/blog/_images/astro-i18n/langpicker.webp"
import githubPulls from "/src/content/blog/_images/astro-i18n/github-pulls.webp"

> ## Welcome, world.

Author: ```suggestion
Welcome, world.
```

Author: Paragraph text instead.

> **Welcome, World.**

Author: ```suggestion
**Leading open-source documentation with and for an international community.**
```

> **Leading open-source documentation with and for an international community.**

Author: ```suggestion
## Welcome, world.
```

## Astro is for everyone

If Astro is truly meant to be for everyone, then you must include the majority of the planet (83%) that does not speak English! 18 months ago, we decided to get serious and invest in overhauling our documentation, and this included a major internationalization (i18n) initiative.

Today, Astro docs has been officially translated into 12 different languages with over 1000 pages translated in total. Even our full introductory tutorial (32 pages in total) has been translated into six different languages. These languages cover billions of people across more than 50 countries, opening the door to many who would otherwise struggle to use Astro reliably.

<BlogContentImage src={langPicker} alt="Astro docs homepage with the language picker drop down menu of available languages" />

Our translations are almost entirely made up of open-source contributions, submitted by volunteers all over the world. We are so, so grateful to everyone who has contributed so far.

> Today is our official Community Day celebration, so I wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and I’d like to share some of our successes — and challenges — with you in this post.

Reviewer 2: There is no I in Co-authored-by.

```suggestion
Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and wanted to share some of our successes — and challenges — with you in this post.
```

> Today is our official Community Day celebration, so I wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and I’d like to share some of our successes — and challenges — with you in this post.

Author: ```suggestion
Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and I’d like to share some of our successes — and challenges — with you in this post.
```

> Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and I’d like to share some of our successes — and challenges — with you in this post.

Author: ```suggestion
Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and we’d like to share some of our successes — and challenges — with you in this post.
```

> Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and we’d like to share some of our successes — and challenges — with you in this post.

Author: ```suggestion
Today is our official Community Day celebration, so we wanted to share how we built this internationalization engine at Astro. We are lucky to have one of the best communities in open-source, and wanted to share some of our successes — and challenges — with you in this post.
```
